Deid GA release

jovinson-ms opened the pull request:
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[azure-health-deidentification](https://apiview.dev/Assemblies/Review/22a62e8622be49bbbd53bc45583d2afa?revisionId=950c5fcf5a974981868dd4a4f3a8d39c)

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[azure-health-deidentification](https://apiview.dev/Assemblies/Review/22a62e8622be49bbbd53bc45583d2afa?revisionId=950c5fcf5a974981868dd4a4f3a8d39c)

jovinson-ms commented:
@davidlaprade we hit some major delays in our SDK GA release but are now picking it back up. I'm still working on updating our READMEs and sample code, but I've incorporated some changes based on Java and .NET SDK reviews that I'd appreciate your feedback on in our APIView. I've updated the changelog so there's an up-to-date summary of changes there as well.
